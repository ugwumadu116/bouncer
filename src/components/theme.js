import * as colors from "./color/color";
import * as fontsize from "./text/fontsize";
import * as fontweight from "./text/fontweight";
import * as lineheight from "./text/lineheight";
import * as letterspacing from "./text/letterspacing";
import * as btnSizes from "./button/button";
import * as btnState from "./button/btnState";
import * as inputsizes from './form/input/inputsizes'
import * as inputbuttonSize from './form/inputButton/inputbuttonsizes'
import * as selectSize from './form/select/selectSizes'


const theme = {
  colors,
  fontsize,
  fontweight,
  lineheight,
  letterspacing,
  btnSizes,
  btnState,
  inputsizes,
  inputbuttonSize,
  selectSize
};
export default theme;
