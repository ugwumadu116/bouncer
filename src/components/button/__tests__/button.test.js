import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import ButtonDom from "../index";

it("renders the default correctly when no props are passed", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where btn size is set to Tiny", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnTiny />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders correctly where btn size is set to small", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnSmall />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where btn size is set to Large", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnLarge />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where inverse is set ", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnInversePrimary />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where transparency is set", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnTransparentPrimary btnSmall />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where background is set primary", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnprimary />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders correctly where background is set Success", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnsuccess />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders correctly where background is set Warning", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btnwarning />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders correctly where background is set Danger", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btndanger />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly where background is set Black", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ButtonDom btndark />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
