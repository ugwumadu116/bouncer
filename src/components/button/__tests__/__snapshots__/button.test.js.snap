// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly where background is set Black 1`] = `
<button
  className="sc-bdVaJa eCAzmP"
/>
`;

exports[`renders correctly where background is set Danger 1`] = `
<button
  className="sc-bdVaJa kgRhys"
/>
`;

exports[`renders correctly where background is set Success 1`] = `
<button
  className="sc-bdVaJa jnlnVZ"
/>
`;

exports[`renders correctly where background is set Warning 1`] = `
<button
  className="sc-bdVaJa cueqap"
/>
`;

exports[`renders correctly where background is set primary 1`] = `
<button
  className="sc-bdVaJa hhpwih"
/>
`;

exports[`renders correctly where btn size is set to Large 1`] = `
<button
  className="sc-bdVaJa fZhNrR"
/>
`;

exports[`renders correctly where btn size is set to Tiny 1`] = `
<button
  className="sc-bdVaJa gLnqv"
/>
`;

exports[`renders correctly where btn size is set to small 1`] = `
<button
  className="sc-bdVaJa hwilJy"
/>
`;

exports[`renders correctly where inverse is set  1`] = `
<button
  className="sc-bdVaJa enqqpa"
/>
`;

exports[`renders correctly where transparency is set 1`] = `
<button
  className="sc-bdVaJa kXpGCg"
/>
`;

exports[`renders the default correctly when no props are passed 1`] = `
<button
  className="sc-bdVaJa hnVdPk"
/>
`;
